### PR TITLE
FAF-21 | Initial TF setup

### PR DESCRIPTION
**Pull request by fwpushan**
----------
1. Set up initial tf
2. Add root make file
3. Add root package

### CODE REVIEW

**Review by matbusby-fw — APPROVED** [review]
I'm not very familiar with terraform, but from what I can tell everything looks good to me.  
I do see a handful of references to nextjs here.  Are those still functional even though we aren't using next?

File: terraform/api.tf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  filename         = var.api_artifact
+  source_code_hash = filebase64sha256(var.api_artifact)
+  handler          = "lambda.handler"
+  memory_size      = var.function_memory_mb
+  timeout          = 30
+
+  environment {
+    variables = {
+      NODE_ENV           = var.target_env
+      AWS_S3_BUCKET      = aws_s3_bucket.uploads.bucket
```
Comment (will-freshworks) [inline terraform/api.tf]: don't need these two AWS_S3 params

File: terraform/cloudfront.tf [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  provider = aws.us-east-1
+  domain   = var.domain
+  statuses = ["ISSUED"]
+}
+
+resource "aws_cloudfront_origin_access_identity" "app" {
+  comment = local.app_name
+}
+
+resource "aws_cloudfront_function" "nextjs_request" {
```
Comment (will-freshworks) [inline terraform/cloudfront.tf]: don't need this cf function any more I think

File: terraform/cloudfront.tf [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+resource "aws_cloudfront_function" "response" {
+  name    = "${local.namespace}-cf-response"
+  runtime = "cloudfront-js-1.0"
+  comment = "Add security headers"
+  code    = file("${path.module}/cloudfront/response.js")
+}
+
+resource "aws_cloudfront_function" "api_request" {
```
Comment (will-freshworks) [inline terraform/cloudfront.tf]: don't need this function either tbh

File: terraform/cloudfront.tf [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    min_ttl     = 0
+    default_ttl = 0
+    max_ttl     = 0
+    compress    = true
+  }
+
+  // Cache _next directory
+  ordered_cache_behavior {
```
Comment (will-freshworks) [inline terraform/cloudfront.tf]: don't need this block

File: terraform/cloudfront.tf [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    viewer_protocol_policy = "redirect-to-https"
+
+    function_association {
+      event_type   = "viewer-response"
+      function_arn = aws_cloudfront_function.response.arn
+    }
+  }
+
+  // Cache img directory
+  ordered_cache_behavior {
```
Comment (will-freshworks) [inline terraform/cloudfront.tf:129]: don't need this block

File: terraform/cloudfront/request.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// Next.js request handler
+function handler(event) {
```
Comment (will-freshworks) [inline terraform/cloudfront/request.js]: don't need this function

File: terraform/cloudfront/api-request.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// add ip address
+function handler(event) {
```
Comment (will-freshworks) [inline terraform/cloudfront/api-request.js]: don't need this function

File: terraform/cloudfront/response.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+function handler(event) {
```
Comment (will-freshworks) [inline terraform/cloudfront/response.js:1]: might as well keep this function

File: terraform/secrets.tf [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# These should be manually populated in the console for each user
+
+data "aws_ssm_parameter" "dynamo_db_password" {
```
Comment (will-freshworks) [inline terraform/secrets.tf:3]: Dynamodb actually doesn't use a password, you do it using lambda permissions

File: terraform/iam.tf [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+          Service = [
+            "lambda.amazonaws.com"
+          ]
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_cloudwatch" {
```
Comment (will-freshworks) [inline terraform/iam.tf:20]: Will need dynamodb access but I can do this as part of helping @kushal-arora-fw get dynamo set-up

**Review by will-freshworks — APPROVED** [review]
Looks good! There's some stuff that can be safely trimmed but you can go ahead and merge if you want and trim that stuff in a later release.

File: terraform/api.tf (hunk 1 above)
Comment (fwpushan) [reply terraform/api.tf]: Removed

File: terraform/iam.tf (hunk 10 above)
Comment (fwpushan) [reply terraform/iam.tf:20]: @will-freshworks  Already added, I will update on SU

File: terraform/secrets.tf (hunk 9 above)
Comment (fwpushan) [reply terraform/secrets.tf:3]: Yes, now I know that.

File: terraform/cloudfront.tf (hunk 2 above)
Comment (fwpushan) [reply terraform/cloudfront.tf]: DONE

File: terraform/cloudfront.tf (hunk 3 above)
Comment (fwpushan) [reply terraform/cloudfront.tf]: DONE

File: terraform/cloudfront.tf (hunk 4 above)
Comment (fwpushan) [reply terraform/cloudfront.tf]: DONE

File: terraform/cloudfront/request.js (hunk 6 above)
Comment (fwpushan) [reply terraform/cloudfront/request.js]: DONE

File: terraform/cloudfront/api-request.js (hunk 7 above)
Comment (fwpushan) [reply terraform/cloudfront/api-request.js]: DONE

**Pull request merged**